include/service/entities/plugin_service.js

```diff
@@ -1116,7 +1116,13 @@ module.exports = function PluginServiceModule(pb) {
             return cb(new Error('PluginService:[INIT] The plugin object must be passed in order to initialize the plugin'), null);
         }
 
-        return new pb.SitePluginInitializationService((PLUGIN_SPECS[plugin.uid]), plugin.site).initialize()
+        let pluginSpec = Promise.resolve(PLUGIN_SPECS[plugin.uid]);
+        if (!pluginSpec) {
+            pluginSpec = new pb.PluginInitializationService(uid).initialize()
```

Comment:
pluginSpec will always be truthy because it is a promise. I don't think the Promise.resolve is necessary but if it is, you'll either have to await or do your initialize in a then handler.